File: Difficulty: Easy/Segregate 0s and 1s/segregate-0s-and-1s.cpp
class Solution {
  public:
    void segregate0and1(vector<int> &arr) {
        // code here
        int j=0;
        int n=arr.size();
        for(;j<n;j++){
            if(arr[j]==1) break;
        }
        
        for(int i=j;i<n;i++){
            if(arr[i]!=1){
                swap(arr[i],arr[j]);
                j++;
            }
        }
        
    }
};

File: Difficulty: Easy/Segregate 0s and 1s/segregate-0s-and-1s.java
class Solution {
    void segregate0and1(int[] arr) {
        // code here
        int n=arr.length;
        int j=0;
        for(;j<n;j++){
            if(arr[j]==1) break;
        }
        
        for(int i=j+1;i<n;i++){
            if(arr[i]==0){
                int temp=arr[i];
                arr[i]=arr[j];
                arr[j]=temp;
                j++;
            }
        }
    }
}
